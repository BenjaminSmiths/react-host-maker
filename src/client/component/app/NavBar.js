import React, {Component} from 'react';
import logo from '../../images/logo.svg';

class NavBar extends Component {

    componentDidMount() {
        window.addEventListener("scroll", () => {
            this.nav.classList.toggle("fix", window.pageYOffset > 70 );
        });
    }

    render() {
        return(<div ref={n => this.nav = n}>
            <div className="NavHeader">
                <div className="headerLogo">
                    <img src={logo} className="App-logo" alt="logo"/>
                </div>
                <div className="fixedHeader">
                    <h2 className="titleText">Welcome to HostMaker</h2>
                </div>
            </div>
        </div>)
    }
}


export default NavBar;