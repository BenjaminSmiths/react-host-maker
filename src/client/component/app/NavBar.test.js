import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from './NavBar';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

describe('NavBar', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NavBar />, div);
    });

    it('should contain the HostMaker SVG', () => {
        // Given
        let component = ReactTestUtils.renderIntoDocument(
            <NavBar />
        );

        // When
        let img = ReactTestUtils.findRenderedDOMComponentWithTag(
            component, 'img'
        );

        // Then
        expect(ReactDOM.findDOMNode(img).src).toEqual("http://localhost/logo.svg");
    });


    it('should say welcome', () => {
        // Given
        let component = ReactTestUtils.renderIntoDocument(
            <NavBar />
        );

        // When
        let h2 = ReactTestUtils.findRenderedDOMComponentWithClass(
            component, 'NavHeader'
        );

        // Then
        expect(ReactDOM.findDOMNode(h2).textContent).toEqual("Welcome to HostMaker");
    });

    it('should match the current snap', () => {

        // When
        const tree = renderer.create(
            <NavBar />
        ).toJSON();

        // Then
        expect(tree).toMatchSnapshot()
    });
    //
    describe('Fixed Header on Scroll', () => {

        it('should add the fix class when offset is over 70', () => {
            // Given
            let component = ReactTestUtils.renderIntoDocument(
                <NavBar />
            );

            // When
            window.pageYOffset = 71;
            window.dispatchEvent(new window.UIEvent('scroll'));

            let fixed = ReactTestUtils.findRenderedDOMComponentWithClass(
                component, 'fix'
            );

            // Then
            expect(fixed).toBeDefined();
        });

        it('should add the fix class when offset is over 70', () => {
            // Given
            let component = ReactTestUtils.renderIntoDocument(
                <NavBar />
            );

            // When
            window.pageYOffset = 0;
            window.dispatchEvent(new window.UIEvent('scroll'));

            let fixed;

            try {
                fixed = ReactTestUtils.findRenderedDOMComponentWithClass(
                    component, 'fix'
                );
            } catch(e) {

            }

            // Then
            expect(fixed).toBeUndefined();
        });

    });

});
