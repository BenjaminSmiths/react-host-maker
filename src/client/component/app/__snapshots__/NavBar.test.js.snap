// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the current snap 1`] = `
<div>
  <div
    className="NavHeader"
  >
    <div
      className="headerLogo"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
    </div>
    <div
      className="fixedHeader"
    >
      <h2
        className="titleText"
      >
        Welcome to HostMaker
      </h2>
    </div>
  </div>
</div>
`;
